app/static/src/app/store/model/actions.ts

```diff
@@ -23,24 +66,20 @@ export const actions: ActionTree<ModelState, AppState> = {
     },
 
     async FetchOdin(context) {
-        const { rootState } = context;
-        const model = rootState.code.currentCode;
+        await fetchOdin(context);
+    },
 
-        await api(context)
-            .withSuccess(ModelMutation.SetOdin)
-            .withError(`errors/${ErrorsMutation.AddError}` as ErrorsMutation, true)
-            .post<OdinModelResponse>("/odin/model", { model });
+    CompileModel(context) {
+        compileModel(context);
     },
 
-    RunModel(context, payload: RunModelPayload) {
-        const { state, commit } = context;
-        if (state.odinRunner && state.odin) {
-            const {
-                parameters, start, end, control
-            } = payload;
+    RunModel(context) {
+        runModel(context);
```

Comment:
```suggestion
    async RunModel(context) {
       await  runModel(context);
```

Reply:
This doesn't actually need to be async as the code itself is completely synchronous (it doesn't make any API requests).

Reply:
The methods are async, as long as the methods are declared async, we should always await. In this case, you may have to remove the async keywords.

Reply:
Ah, sorry, i have erroneously marked those functions as async when they don't need to be. My mistake.